src/NuGet.Core/NuGet.Versioning/VersionRangeFactory.cs

```diff
@@ -116,6 +116,17 @@ public static bool TryParse(string value, bool allowFloating, out VersionRange v
                 return false;
             }
 
+            // Special case *-*
+            if (allowFloating
+                && charArray.Length == 3
+                && charArray[0] == '*'
```

Comment:
Pretty sure you intended the array indexes to be 0,1,2, respectively.

Reply:
Ahh, my bad, I thought I removed that. 

As this is currently, it's dead code, and there are tests covering this scenario, so removing the special case is safe, so I'll just do that.

Reply:
Apparently I left it in the FloatRange, so I'll just do the same.

Reply:
Removed.